Automattic/newspack-newsletters
# Add layouts management

### All Submissions:






### Changes proposed in this Pull Request:



Enables managing layouts.

Closes #31 
Closes #216

### How to test the changes in this Pull Request:

1. Create a new newsletter – observe two tabs in the layout selection pane: "Default layouts", "My layouts"
2. Observe that the latter tab is empty and displaying a message about new layout creation
3. Choose one of the default layouts to create a newsletter 
3. Observe a new "Create a new layout" button in the "Layout" section of the sidebar Document settings, click it
3. Observe a modal displayed with new layout name input. It should be set to the newsletter title 
3. Click "Save" 
3. Create a new newsletter. Observe that the new layout is listen in the "My layouts" tab
3. Select the new layout to create a new newsletter
3. Edit the newsletter content
3. Observe a "Update layout" button besides the "Create a new layout" button in the "Layout" section of the sidebar Document settings, click it
3. Change the name of the layout and save it
3. Add new newsletter. Observe the layout content & title have been updated
3. Click "Delete" button below the layout tile in the selection tab, confirm
3. Observe the layout has been deleted

### Other information:
